x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/action_log_expanded_tray.tsx

```diff
@@ -0,0 +1,221 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo } from 'react';
+import { EuiCodeBlock, EuiFlexGroup, EuiFlexItem, EuiDescriptionList } from '@elastic/eui';
+import { css, euiStyled } from '@kbn/kibana-react-plugin/common';
+import { i18n } from '@kbn/i18n';
+import { useUserPrivileges } from '../../../../common/components/user_privileges';
+import { OUTPUT_MESSAGES } from '../translations';
+import { getUiCommand } from './hooks';
+import { useTestIdGenerator } from '../../../hooks/use_test_id_generator';
+import { ResponseActionFileDownloadLink } from '../../response_action_file_download_link';
+import { ExecuteActionHostResponseOutput } from '../../endpoint_execute_action';
+import { getEmptyValue } from '../../../../common/components/empty_value';
+
+import { type ActionDetails, type MaybeImmutable } from '../../../../../common/endpoint/types';
+const EXECUTE_FILE_LINK_TITLE = i18n.translate(
+  'xpack.securitySolution.responseActionExecuteDownloadLink.downloadButtonLabel',
+  { defaultMessage: 'Click here to download full output' }
+);
+const emptyValue = getEmptyValue();
+
+const customDescriptionListCss = css`
+  &.euiDescriptionList {
+    > .euiDescriptionList__title {
+      color: ${(props) => props.theme.eui.euiColorDarkShade};
+      font-size: ${(props) => props.theme.eui.euiFontSizeXS};
+    }
+
+    > .euiDescriptionList__title,
+    > .euiDescriptionList__description {
+      font-weight: ${(props) => props.theme.eui.euiFontWeightRegular};
+      margin-top: ${(props) => props.theme.eui.euiSizeS};
+    }
+  }
+`;
+const topSpacingCss = css`
+  ${(props) => `${props.theme.eui.euiSize} 0`}
+`;
+const dashedBorderCss = css`
+  ${(props) => `1px dashed ${props.theme.eui.euiColorDisabled}`};
+`;
+const StyledDescriptionListOutput = euiStyled(EuiDescriptionList).attrs({ compressed: true })`
+  ${customDescriptionListCss}
+  dd {
+    margin: ${topSpacingCss};
+    padding: ${topSpacingCss};
+    border-top: ${dashedBorderCss};
+    border-bottom: ${dashedBorderCss};
+  }
+`;
+
+const StyledDescriptionList = euiStyled(EuiDescriptionList).attrs({
+  compressed: true,
+  type: 'column',
+})`
+  ${customDescriptionListCss}
+`;
+
+const StyledEuiCodeBlock = euiStyled(EuiCodeBlock).attrs({
+  transparentBackground: true,
+  paddingSize: 'none',
+})`
+  code {
+    color: ${(props) => props.theme.eui.euiColorDarkShade} !important;
+  }
+`;
+
+export const ActionsLogExpandedTray = memo<{
+  action: MaybeImmutable<ActionDetails>;
+}>(({ action }) => {
+  const getTestId = useTestIdGenerator('response-actions-list');
+  const { canWriteFileOperations, canWriteExecuteOperations } =
+    useUserPrivileges().endpointPrivileges;
+
+  const {
+    startedAt,
+    completedAt,
+    isCompleted,
+    wasSuccessful,
+    isExpired,
+    command: _command,
+    comment,
+    parameters,
+  } = action;
+
+  const parametersList = parameters
```

Comment:
can you memoize this

Reply:
7a1f87b1bee40512a7d11b44ad16cd3d781748c6